homeassistant/components/keenetic_ndms2/__init__.py

```diff
@@ -65,6 +69,24 @@ async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry) ->
 
     hass.data[DOMAIN].pop(config_entry.entry_id)
 
+    if router.tracked_interfaces - set(config_entry.options[CONF_INTERFACES]):
+        _LOGGER.debug(
+            "Removing device_tracker entities since some interfaces are now untracked"
+        )
+        ent_reg = await entity_registry.async_get_registry(hass)
+        dev_reg = await device_registry.async_get_registry(hass)
+        for entity_entry in list(ent_reg.entities.values()):
+            if (
+                entity_entry.config_entry_id == config_entry.entry_id
+                and entity_entry.domain == DEVICE_TRACKER_DOMAIN
+            ):
+                _LOGGER.debug("Removing entity %s", entity_entry.entity_id)
+
+                ent_reg.async_remove(entity_entry.entity_id)
+                dev_reg.async_update_device(
+                    entity_entry.device_id, remove_config_entry_id=config_entry.entry_id
+                )
```

Comment:
Just wondering if we should clean this up on setup instead of unload?
I think we should do the least possible on e.g., Home Assistant shutdown.

Reply:
@frenck Unfortunately on setup the Router instance is already destroyed